test/lib/rules/prefer-custom-event.js

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+'use strict';
+
+const { RuleTester } = require('eslint');
+
+const { ESLINT_TEST_CONFIG } = require('../shared');
+const rule = require('../../../lib/rules/prefer-custom-event');
+
+const ruleTester = new RuleTester(ESLINT_TEST_CONFIG);
+
+ruleTester.run('prefer-custom-event', rule, {
+    valid: [
+        {
+            code: `new CustomEvent('test');`,
+        },
+        {
+            code: `
+                class Event {}
+                new Event('test');
+            `,
+        },
+    ],
+    invalid: [
+        {
+            code: `new Event('test');`,
+            errors: [
+                {
+                    message:
+                        'Prefer using "CustomEvent" constructor over "Event" for dispatching events.',
+                },
+            ],
+        },
+        {
+            code: `new Event('test');`,
+            env: {
+                browser: true,
+            },
+            errors: [
+                {
+                    message:
+                        'Prefer using "CustomEvent" constructor over "Event" for dispatching events.',
+                },
+            ],
+        },
```

Comment:
Thought of one other invalid case which developers can use to by-pass:
`new Event.prototype.constructor('test');`

There are more ways to bypass this restriction, but I thought this is an easy one and most likely to be used. Not a deal breaker if you don't think it is necessary to stop developers who really want to bypass the restriction.

Reply:
I feel the same way, developers might work around this restriction by using `window.Event` as well. Since this rule is really a suggestion more than a restriction, I prefer to keep it simple.